scripts/deploymentextensions/enablemobileapi/install.sh

```diff
@@ -0,0 +1,394 @@
+if [[ $debug_mode == 1 ]];
+then
+    set -x
+fi
+
+# Sync the oxa-tools repository (with tag support)
+repo_url=`get_github_url "$oxa_tools_public_github_account" "$oxa_tools_public_github_projectname"`
+sync_repo $repo_url $oxa_tools_public_github_projectbranch $oxa_tools_repository_path $access_token $oxa_tools_public_github_branchtag
+log "Repository sync is complete" " " 2
+
+####################################
+# Main Operations
+####################################
+
+if [[ $remote_mode == 1 ]];
+then
+    # This phase ONLY executes on the remote target
+
+    # Conditionally install jq
+    install-json-processor
+
+    # Update the target lms environment file (setting to true if false or creating a new feature that is enabled)
+    target_file="/edx/app/edxapp/lms.env.json"
+
+    # backup the existing config file
+    log "Backing up the configuration file: ${target_file}"
+    cp $target_file{,.backup}
+
+    # get an temporary copy & apply modifications
+    temp_file=`mktemp "/tmp/lms.env.json.XXXXXXXXXX"`
+
+    log "Enabling OAUTH2 & Mobile Rest Api"
+    cat "${target_file}" | jq -r '.FEATURES.ENABLE_OAUTH2_PROVIDER |= true | .FEATURES.OAUTH_ENFORCE_SECURE |= true | .FEATURES.ENABLE_MOBILE_REST_API |= true' --indent 4 > $temp_file
+    exit_on_error "Could not enable oauth2 & mobile rest api on ${HOSTNAME}!" "${error_mobilerestapi_update_failed}" "${notification_email_subject}" "${cluster_admin_email}"
+
+    # move the temporary copy as the new target file
+    log "Moving the updated configuration file from '${temp_file}' to '${target_file}'"
+    cp $temp_file $target_file
+
+    # make sure the permissions remain in tact
+    chown edxapp:www-data "${target_file}"
+    chmod 644 "${target_file}"
+
+    # restart the services
+    log "Restarting services on ${HOSTNAME}"
+    /edx/bin/supervisorctl restart all
+    exit_on_error "Could not restart all services on ${HOSTNAME}!" "${error_mobilerestapi_update_failed}" "${notification_email_subject}" "${cluster_admin_email}"
+
+    log "Sleeping ${wait_interval_seconds} seconds before moving to update the next server"
+    sleep "${wait_interval_seconds}s"
+
+    log "Updated ${HOSTNAME} successfully"
+    exit
+fi
+
+# login
+authenticate-azureuser
+
+# List the VMSSs in the target cluster. 
+# If the user specifies a VMSS filter (the deploymentId), list only that VMSS. Otherwise, list all VMSSs.
+log "Getting the list of VMSS in the '${azure_resource_group}' resource group"
+vmssIdsArray=(`az vmss list --resource-group ${azure_resource_group} | jq -r '.[] | .id'`)
+exit_on_error "Could not get a list of the VMSSs in the '${azure_resource_group}' resource group from '${HOSTNAME}' !" "${error_mobilerestapi_update_failed}" "${notification_email_subject}" "${cluster_admin_email}"
+
+vmssIdsList=""
+for vmssId in "${vmssIdsArray[@]}"
+do
+    if [[ -z $vmss_deployment_id ]] || ( [[ ! -z $vmss_deployment_id ]] && [[ $vmssId == *"vmss-$vmss_deployment_id" ]] );
```

Comment:
for readability: negation in conditionals should be avoided especially cases where it's easy to do. a lot of coding style guides adopt this pattern. if it's helpful, here's a commonly used example to illustrate the point
if a > b
is far better and more readable than the equivalent
if ! a <= b

another popular example
if isEven()
is better than
if  ! IsOdd()